mlflow/utils/thread_utils.py

```diff
@@ -1,36 +1,46 @@
 import os
 import threading
 
-_thread_locals = threading.local()
-_globals = {}
-_globals_lock = threading.Lock()
 
-
-def set_thread_local_var(key, value):
-    """
-    Set a thread-local variable.
+class ThreadLocalVariable:
     """
-    global _thread_locals
-    setattr(_thread_locals, key, (value, os.getpid()))
+    Class for creating a thread local variable.
 
-
-def get_thread_local_var(key, init_value, reset_in_subprocess=True):
-    """
-    Get a thread-local variable.
-    If the thread-local variable is not set, return the provided `init_value` value.
-    If `get_thread_local_var` is called in a forked subprocess and `reset_in_subprocess` is True,
-    return the provided `init_value` value
+    Args:
+        default_factory: A function used to create the default value
+        reset_in_subprocess: Indicating whether the variable is reset in subprocess.
     """
-    global _thread_locals
 
-    if hasattr(_thread_locals, key):
-        value, pid = getattr(_thread_locals, key)
-        if reset_in_subprocess and pid != os.getpid():
-            # `get_thread_local_var` is called in a forked subprocess, reset it.
-            set_thread_local_var(key, init_value)
-            return init_value
+    def __init__(self, default_factory, reset_in_subprocess=True):
+        self.reset_in_subprocess = reset_in_subprocess
+        self.default_factory = default_factory
+        self.thread_local = threading.local()
+        self._value_dict = {}
```

Comment:
What is this?

Reply:
_value_dict is used in `mlflow/spark/autologging.py` to address spark datasource autologging issue.

Reply:
Can you add a comment? It's unclear how it should be used.

Reply:
Sure.